lib/src/adder.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// carry_save_multiplier.dart
+// Implementation of pipeline multiplier module.
+//
+// 2023 May 29
+// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+
+/// An abstract class for all adder module.
+abstract class Adder extends Module {
+  /// The List of inputs to sum. Length of inputs must be two.
+  List<Logic> toSum;
+
+  /// The addition results [sum].
+  List<Logic> get sum;
```

Comment:
The output of an adder should be a `Logic`, not a `List<Logic>`.  Swizzle it in implementations.